src/SFA.DAS.QnA.Application/Commands/ResetSectionAnswers/ResetSectionAnswersHandler.cs

```diff
@@ -0,0 +1,62 @@
+﻿using MediatR;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.QnA.Api.Types;
+using SFA.DAS.QnA.Application.Commands.Files.DeleteFile;
+using SFA.DAS.QnA.Application.Commands.SetPageAnswers;
+using SFA.DAS.QnA.Application.Services;
+using SFA.DAS.QnA.Data;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.QnA.Application.Commands.ResetPageAnswers
+{
+    public class ResetSectionAnswersHandler : SetAnswersBase, IRequestHandler<ResetSectionAnswersRequest, HandlerResponse<ResetSectionAnswersResponse>>
+    {
+        private readonly IMediator _mediator;
+
+        public ResetSectionAnswersHandler(QnaDataContext dataContext, IMediator mediator, INotRequiredProcessor notRequiredProcessor, ITagProcessingService tagProcessingService) 
+            : base(dataContext, notRequiredProcessor, tagProcessingService, null)
+        {
+            _mediator = mediator;
+        }
+
+        public async Task<HandlerResponse<ResetSectionAnswersResponse>> Handle(ResetSectionAnswersRequest request, CancellationToken cancellationToken)
+        {
+            var application = await _dataContext.Applications.SingleOrDefaultAsync(app => app.Id == request.ApplicationId, cancellationToken);
```

Comment:
I'd move this to being below the validationErrorResponse so you're not wasting a database call if validation fails

Reply:
Done